feat: adding cat painting project

Ksound22 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

<!-- Feel free to add any additional description of changes below this line -->

This is the project that will replace the Picasso painting project.

Ksound22 committed:
feat: finished breaking cat painting project into steps

Ksound22 committed:
ci: fixing failing CI

Ksound22 committed:
ci: chnaged order to make CI pass

moT01 commented:
Is this ready to go or are you still working on it @Ksound22?

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/6476fc5cf14b276e6d04e82a.md:
```diff
@@ -0,0 +1,67 @@
+
+# --description--
+
+Use class selector to give your `.box` element a width of `200px`, a height of `600px`, and a background color of `#000`. Also, give it a `position` of `absolute`, a `left` of `650px`, and a `top` of `800px`.
```
```suggestion
Use a class selector to give your `.box` element a width of `200px`, a height of `600px`, and a background color of `#000`. Also, give it a `position` of `absolute`, a `left` of `650px`, and a `top` of `800px`.
```

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/6476fd4213318f6ee211028a.md:
```diff
@@ -0,0 +1,75 @@
+
+# --description--
+
+Now replace the position property value of your `.cat-head` with `fixed`. Leave both `top` and `left` as they are.
```
```suggestion
Now replace the `position` property value of your `.cat-head` with `fixed`. Leave both `top` and `left` as they are.
```

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/6477062778c85972eb648030.md:
```diff
@@ -0,0 +1,76 @@
+
+Give the `cat-head` element a position property set to `absolute`. Then set a value of `0` for the `right`, `left`, `top`, `bottom` properties, then set its `margin` property on all sides to `auto`. That's one way to center an element vertically and horizontally using CSS positioning.
+
+You should also remove the `.box` CSS rule and its declarations because you don't need them anymore.
```
If we are asking them to remove the box rule, it should be in the editable region.

jdwilkin4 commented on curriculum/challenges/_meta/learn-intermediate-css-by-building-a-cat-painting/meta.json:
```diff
@@ -0,0 +1,357 @@
+  "usesMultifileEditor": true,
+  "hasEditableBoundaries": true,
+  "dashedName": "learn-intermediate-css-by-building-a-cat-painting",
+  "order": 20,
```
According to the spreadsheet, this project is listed right after the picasso painting. So that would mean the order is supposed to be 12 and not 20. 

But my understanding was this project is meant to replace the picasso paining. 
Is that right?

If so, then I guess the number should be 11. 

I would reach out to the team about that.

naomi-lgbt replied:
I'll audit the order once all of the projects are in.

Ksound22 replied:
@naomi-lgbt and @jdwilkin4, can I go ahead and resolve this conversation?

naomi-lgbt replied:
I was incorrect - this is the RWD not the JS, which means it's already on production. What we'll need to do is use the correct order in this PR:

```suggestion
  "order": 11,
```

Then once this is merged, merge #50768 before deploying.

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c47867800472a4ed5d2ea.md:
```diff
@@ -0,0 +1,28 @@
+
+# --description--
+
+Start by setting up an HTML boilerplate. Add a `<!DOCTYPE>` of `html` and an `html` element with a `lang` attribute set to `en`. Within your `html` element, add a `head` element and a `body` element.
```
I like the wording from the building a piano project better here

```suggestion
Begin with the basic HTML structure. Add a `DOCTYPE` declaration and `html`, `head`, and `body` elements.
```

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c48df8674cf2b91020ecc.md:
```diff
@@ -0,0 +1,38 @@
+
+# --description--
+
+Even though you have not written any CSS yet, you can link your CSS file now.
+
+Add a `link` element with a `rel` attribute of `stylesheet`, and an `href` of `styles.css`.
```
I prefer the wording from the build a piano project here

```suggestion
Add a `link` element within your `head` element. For that `link` element, set the `rel` attribute to `stylesheet` and the `href` to `./styles.css`.
```

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c586be7180e40ddf74ff6.md:
````diff
@@ -0,0 +1,47 @@
+```
+
+```css
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
````
@naomi-lgbt 
For the new JS projects, some of the css files do have this minimal css reset here.

But for this particular certification, I do think we should stay consistent with what the other projects are doing.

After a quick glance it seems like most of the other projects are setting the `box-sizing` for the html element.

```css
html {
  box-sizing: border-box;
}
```
@Ksound22 I vote we use that instead

Sboonny replied:
```css
html {
  box-sizing: border-box;
}
```

`box-sizing` isn't inherited by default, if there are projects that use the syntax above, it needs to be looked at if they actually need that property, so this issue doesn't repeat itself https://github.com/freeCodeCamp/freeCodeCamp/issues/49284

jdwilkin4 replied:
@Sboonny 

After looking more closely at all of the practice projects there doesn't seem to be a consistent use of where the box-sizing is set. 


The balance sheet project and the building a piano project set the box-sizing for the html element.

```
html {
  box-sizing: border-box;
}
```
The photo gallery, nutrition label, and city skyline set the box sizing to the `*` selector. 

```
* {
  box-sizing: border-box;
}
```
If this cat painting project falls after the magazine project, then we can just leave the current implementation as it is since campers have already been taught that. 

Otherwise, we could just apply the box sizing to the `*` selector

Ksound22 replied:
> Why are we writing this for the campers?

Just a few resets I always put in my CSS. I'll do what Jessica suggested about it.

Ksound22 replied:
> @Sboonny
> 
> After looking more closely at all of the practice projects there doesn't seem to be an inconsistent use of where the box-sizing is set.
> 
> The balance sheet project and the building a piano project set the box-sizing for the html element.
> 
> ```
> html {
>   box-sizing: border-box;
> }
> ```
> 
> The photo gallery, nutrition label, and city skyline set the box sizing to the `*` selector.
> 
> ```
> * {
>   box-sizing: border-box;
> }
> ```
> 
> If this cat painting project falls after the magazine project, then we can just leave the current implementation as it is since campers have already been taught that.
> 
> Otherwise, we could just apply the box sizing to the `*` selector

Which should I use?
```css
html {
   box-sizing: border-box;
```
or

```css
* {
   box-sizing: border-box;
}
```

jdwilkin4 replied:
@Ksound22 

I would just go with using the `*` selector. 

```
* {
   box-sizing: border-box;
}
```
You will also need to create a new step before this one to have campers add it. 

You can use the Insert step option mentioned [here](https://contribute.freecodecamp.org/#/how-to-work-on-practice-projects?id=insert-step) in the docs to do that.

jdwilkin4 reviewed (changes requested):
I think it is off to a good start.

Left a few comments 👍

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/6476f7a4827bcc61682f2347.md:
```diff
@@ -0,0 +1,67 @@
+
+Set the position property of your `.cat-head` element to `absolute`, then set `top` and `left` to `300px` and `400px` respectively.
+
+**Note**: You can experiment with `top`, `left`, `bottom`, and `right` properties here, but the test would only pass for `top` of `300px`, and left of `400px`.
```
Ok. I'll look into that.

Ksound22 committed:
chore: implemented Jessica's suggestions

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

Ksound22 committed:
chore: implemented Naomi's review suggestions

Co-authored-by: Naomi Carrigan <nhcarrigan@gmail.com>

Ksound22 committed:
fix: implemented Naomi's suggestion on putting the box rule in the editable area

Ksound22 committed:
fix: removed two lines of existing CSS resets

jdwilkin4 commented on client/src/pages/learn/2022/responsive-web-design/learn-intermediate-css-by-building-a-cat-painting/index.md:
```diff
@@ -0,0 +1,10 @@
+title: Introduction to the Learn Intermediate CSS by Building a Cat Painting
+block: learn-intermediate-css-by-building-a-cat-painting
+superBlock: 2022/responsive-web-design
+isBeta: true
```
This shouldn't be in beta since the responsive web design curriculum is already live

Ksound22 committed:
feat: finished writing tests

Ksound22 committed:
fix: resolved CI issues

Ksound22 committed:
feat: rewrote tests for some steps

ojeytonwilliams commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c47867800472a4ed5d2ea.md:
```diff
@@ -0,0 +1,82 @@
+
+# --description--
+
+Begin with the basic HTML structure. Add a `DOCTYPE` reference of `html` and an `html` element with its `lang` attribute set to `en`. Also, add a `head` and `body` elements within the `html` element.
```
```suggestion
Begin with the basic HTML structure. Add a `DOCTYPE` reference of `html` and an `html` element with its `lang` attribute set to `en`. Also, add a `head` and a `body` element within the `html` element.
```
Grammar nit

ojeytonwilliams reviewed (dismissed):
Hey, this isn't a proper review (though it does look good!) I just noticed a typo when I glanced at this.

zairahira reviewed (changes requested):
Went through all the steps. The tests are working perfectly fine🚀
I have left some comments where the description needs to match the tests.

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f516dbfc1342495515625.md:
```diff
@@ -0,0 +1,572 @@
+
+# --description--
+
+Rotate the bottom-right whisker at 3 degrees.
```
Since this is the last step, we should add a short message at the end acknowledging that. Just like the other projects do.

```suggestion
Rotate the bottom-right whisker at 3 degrees.

With this final step, your cat painting is now complete.
```

jdwilkin4 reviewed (changes requested):
I went through all of the steps and tests and I think it looks good.
Left a couple of small comments 👍

Ksound22 committed:
chore: implemented review suggestions from Oliver, Zaira, and Jessica.

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>
Co-authored-by: Zaira <33151350+zairahira@users.noreply.github.com>
Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f164bf100dd11d226161f.md:
```diff
@@ -0,0 +1,256 @@
+
+# --description--
+
+Rotate the right middle whisker at 3 degrees.
```
```suggestion
Rotate the left middle whisker at 3 degrees.
```

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f1802a09a171332e14630.md:
```diff
@@ -0,0 +1,266 @@
+
+# --description--
+
+Rotate the left bottom whisker at -3 degree.
```
```suggestion
Rotate the left bottom whisker at -3 degrees.
```

naomi-lgbt reviewed (changes requested):
A couple of nitpicks in the suggestions below, but I have some larger thoughts:

First, and I wish I had caught this when this was added to the curriculum expansion repo... There is a *lot* of repeated CSS. I feel like we're missing an opportunity to reinforce applying styles to multiple classes at once. It would be a pretty big refactor to change that at this point, but IMHO it's worth it. (It also felt tedious as a learner to be writing the same lines so many times)

Second, the indentation for blank lines in the editable regions are inconsistent. I've had some that started with 4 indents when they should only have 2 - even one with a good 10 or so indents. IMHO, new lines we provide for campers to write their code should either start at the appropriate indent level, or no indentation at all.

Ksound22 committed:
chore: implemented Naomi's suggestions on indentation

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c5ffef5598d449b52ec12.md:
```diff
@@ -0,0 +1,80 @@
+</head>
+<body>
+    <main>
+      --fcc-editable-region--
+      <div class="cat-head">
+        
+      </div>
+      --fcc-editable-region--
```
The previous step is about removing it. There's a test for it.

Ksound22 replied:
I will insert a step for it

Ksound22 commented:
> A couple of nitpicks in the suggestions below, but I have some larger thoughts:
> 
> First, and I wish I had caught this when this was added to the curriculum expansion repo... There is a _lot_ of repeated CSS. I feel like we're missing an opportunity to reinforce applying styles to multiple classes at once. It would be a pretty big refactor to change that at this point, but IMHO it's worth it. (It also felt tedious as a learner to be writing the same lines so many times)
> 
> Second, the indentation for blank lines in the editable regions are inconsistent. I've had some that started with 4 indents when they should only have 2 - even one with a good 10 or so indents. IMHO, new lines we provide for campers to write their code should either start at the appropriate indent level, or no indentation at all.

I have fixed the indentation part. How would you suggest I refactor the whole code to reduce the repeated CSS?

Ksound22 committed:
chore: implemented Naomi's suggestions

Co-authored-by: Naomi Carrigan <nhcarrigan@gmail.com>

Ksound22 committed:
chore: added a new step to remove box div

naomi-lgbt commented:
> I have fixed the indentation part. How would you suggest I refactor the whole code to reduce the repeated CSS?

It would be a pretty big refactor at this point. Might not be worth doing, but essentially the common properties (e.g. width and height for all six whiskers) should be moved to a separate multi-class selector.

gikf commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646ce8bb4b35544d501c7184.md:
```diff
@@ -0,0 +1,86 @@
+
+# --description--
+
+Inside your `.cat-ears` element, create two `div` elements with the classes `left-ear` and `right-ear` respectively.
```
`left-ear` and `right-ear` class names are inconsistent with naming of other classes which always have `cat-` prefix.

Ksound22 replied:
Thank you. I'll change it.

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/6476f7a4827bcc61682f2347.md:
```diff
@@ -0,0 +1,79 @@
+
+Set the position property of your `.cat-head` element to `absolute`, then set `top` and `left` properties to any pixel value.
+
+<!-- **Note**: You can experiment with `top`, `left`, `bottom`, and `right` properties here, but the test would only pass for `top` of `300px`, and left of `400px`. -->
```
Are you saying the seed in the next step for the pixel values should be the same as the ones here?

gikf replied:
What I meant was uncommenting the instruction and test, to require `top: 300px` and `left: 400px` to complete step 11. But changing values in next step also would work.

Currently step 11 can be completed with:
```
//Step 11
  top: 100px;
  left: 100px;
```
However next step will start with seed having:
```
//Step 12
  top: 300px;
  left: 400px;
```
Which can be confusing.

Ksound22 replied:
Step 11 lets the learners add in any value for the `top` and `left` properties, which I think is better than telling them the tests would only pass for a `top` of `300px` and a `left` of `400px`.

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f102bf87b350b593baa72.md:
```diff
@@ -0,0 +1,258 @@
+
+        <div class="cat-whiskers">
+          <div class="cat-whiskers-left">
+             <div class="cat-whisker-left-top"></div>
+              <div class="cat-whisker-left-middle"></div>
+              <div class="cat-whisker-left-bottom"></div>
```
I will check the others

Ksound22 committed:
chore: implemented review suggestions from Krzysztof

Co-authored-by: Krzysztof G. <60067306+gikf@users.noreply.github.com>

Ksound22 committed:
chore: fixed HTML indentation issues as suggested by Krzysztof

Ksound22 committed:
chore: prefixed left-ear and right-ear classes with 'cat-'

Ksound22 commented:
> > I have fixed the indentation part. How would you suggest I refactor the whole code to reduce the repeated CSS?
> 
> It would be a pretty big refactor at this point. Might not be worth doing, but essentially the common properties (e.g. width and height for all six whiskers) should be moved to a separate multi-class selector.

This works fine on my end:

```css
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background-color: #c9d2fc;
}

.cat-head {
  position: absolute;
  right: 0;
  left: 0;
  top: 0;
  bottom: 0;
  margin: auto;
  background: linear-gradient(#5e5e5e 85%, #45454f 100%);
  width: 205px;
  height: 180px;
  border: 1px solid #000;
  border-radius: 46%;
}

.left-ear {
  position: absolute;
  top: -26px;
  left: -31px;
  z-index: 1;
  transform: rotate(-45deg);
  border-top-left-radius: 90px;
  border-top-right-radius: 10px;
  border-left: 35px solid transparent;
  border-right: 35px solid transparent;
  border-bottom: 70px solid #5e5e5e;
}

.right-ear {
  position: absolute;
  top: -26px;
  left: 163px;
  z-index: 1;
  transform: rotate(45deg);
  border-top-left-radius: 90px;
  border-top-right-radius: 10px;
  border-left: 35px solid transparent;
  border-right: 35px solid transparent;
  border-bottom: 70px solid #5e5e5e;
}

.cat-left-inner-ear {
  position: absolute;
  top: 22px;
  left: -20px;
  border-top-left-radius: 90px;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 40%;
  border-bottom-left-radius: 40%;
  border-left: 20px solid transparent;
  border-right: 20px solid transparent;
  border-bottom: 40px solid #3b3b4f;
}

.cat-right-inner-ear {
  position: absolute;
  top: 22px;
  left: -20px;
  border-top-left-radius: 90px;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 40%;
  border-bottom-left-radius: 40%;
  border-left: 20px solid transparent;
  border-right: 20px solid transparent;
  border-bottom: 40px solid #3b3b4f;
}

.cat-left-eye {
  position: absolute;
  top: 54px;
  left: 39px;
  transform: rotate(25deg);
  width: 30px;
  height: 40px;
  background-color: #000;
  border-radius: 60%;
}

.cat-right-eye {
  position: absolute;
  top: 54px;
  left: 134px;
  transform: rotate(-25deg);
  width: 30px;
  height: 40px;
  background-color: #000;
  border-radius: 60%;
}

.cat-left-inner-eye {
  position: absolute;
  top: 8px;
  left: 2px;
  transform: rotate(10deg);
  width: 10px;
  height: 20px;
  background-color: #fff;
  border-radius: 60%;
}

.cat-right-inner-eye {
  position: absolute;
  top: 8px;
  left: 18px;
  transform: rotate(-5deg);
  width: 10px;
  height: 20px;
  background-color: #fff;
  border-radius: 60%;
}

.cat-nose {
  position: absolute;
  top: 108px;
  left: 85px;
  border-top-left-radius: 50%;
  border-bottom-right-radius: 50%;
  border-bottom-left-radius: 50%;
  transform: rotate(180deg);
  border-left: 15px solid transparent;
  border-right: 15px solid transparent;
  border-bottom: 20px solid #442c2c;
}

/* New multi-level selector for mouth starts */
.cat-mouth div {
  width: 30px;
  height: 50px;
  border: solid 2px #000;
  border-radius: 190%/190px 150px 0 0;
  border-color: black transparent transparent transparent;
}
/* New multi-level selector for mouth ends */

.cat-mouth-line-left {
  position: absolute;
  top: 88px;
  left: 74px;
  transform: rotate(170deg);
}

.cat-mouth-line-right {
  position: absolute;
  top: 88px;
  left: 91px;
  transform: rotate(165deg);
}

/* New multi-level selector for whiskers starts */
.cat-whiskers-left div {
  width: 40px;
  height: 1px;
  background-color: #000;
}

.cat-whiskers-right div {
  width: 40px;
  height: 1px;
  background-color: #000;
}
/* New multi-level selector ends */

.cat-whisker-left-top {
  position: absolute;
  top: 120px;
  left: 52px;
  transform: rotate(10deg);
}

.cat-whisker-left-middle {
  position: absolute;
  top: 127px;
  left: 52px;
  transform: rotate(3deg);
}

.cat-whisker-left-bottom {
  position: absolute;
  top: 134px;
  left: 52px;
  transform: rotate(-3deg);
}

/* Right */
.cat-whisker-right-top {
  position: absolute;
  top: 120px;
  left: 109px;
  transform: rotate(-10deg);
}

.cat-whisker-right-middle {
  position: absolute;
  top: 127px;
  left: 109px;
  transform: rotate(-3deg);
}

.cat-whisker-right-bottom {
  position: absolute;
  top: 134px;
  left: 109px;
  transform: rotate(3deg);
}

```

Apart from the whiskers, I also moved common selectors of the mouth to a multi-level selector. Is that enough?

Ksound22 committed:
chore: implemented a part of Naomi's suggestion on removing repeated CSS s

Ksound22 committed:
chore: refactored CSS to in order to remove some more repeated CSS

Ksound22 commented:
> > I have fixed the indentation part. How would you suggest I refactor the whole code to reduce the repeated CSS?
> 
> It would be a pretty big refactor at this point. Might not be worth doing, but essentially the common properties (e.g. width and height for all six whiskers) should be moved to a separate multi-class selector.

@naomi-lgbt could you please take another look now? I have moved those properties into some multi-class selectors.

Dario-DC commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646de8478d6f796bfbdccfb2.md:
````diff
@@ -0,0 +1,162 @@
+assert(new __helpers.CSSHelp(document).getStyle('.cat-right-eye')?.borderRadius === '60%')
+```
+
+Your `.cat-right-eye` selector should have a `transform` property set to `-25deg`.
````
I would specify "set to `rotate(-25deg)`".

Dario-DC commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646de5dc8988076a1d992afd.md:
````diff
@@ -0,0 +1,150 @@
+assert(new __helpers.CSSHelp(document).getStyle('.cat-left-eye')?.borderRadius === '60%')
+```
+
+Your `.cat-left-eye` selector should have a `transform` property set to `25deg`.
````
I would specify "set to `rotate(25deg)`".

Dario-DC reviewed (commented):
I went through all the steps and left a few comments. It looks nice.

zairahira commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646ce8bb4b35544d501c7184.md:
```diff
@@ -0,0 +1,86 @@
+
+# --description--
+
+Inside your `.cat-ears` element, create two `div` elements with the classes `cat-left-ear` and `cat-right-ear` respectively.
```
Including check for closing `div` tags:


    Make sure to close the div tags.
    
     ```js
      assert.match(code, /<\/div>/g).length === 3);
    
     ```

zairahira reviewed (commented):
Reviewed till step 20.
A quick question- are we checking the semicolons in CSS?

Ksound22 commented:
> Reviewed till step 20. A quick question- are we checking the semicolons in CSS?

No we are not.

zairahira reviewed (commented):
Reviewed till step 30.

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/6476f7a4827bcc61682f2347.md:
```diff
@@ -0,0 +1,79 @@
+
+# --description--
+
+The next position property is `absolute`. When you use the `absolute` value for your `position` property, it is taken out of the normal flow of the document, and then its position is determined by the `top`, `right`, `bottom`, and `left` properties.
```
Thank you

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c5ffef5598d449b52ec12.md:
````diff
@@ -0,0 +1,80 @@
+Your `div` element should have the class `cat-ears`.
+
+```js
+assert(document.querySelectorAll('.cat-head div')[0]?.getAttribute('class') === 'cat-ears');
````
It happens here too: https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-intermediate-css-by-building-a-picasso-painting/step-11 – a production project BTW.

What if it's due to what @ShaunSHamilton explained a while ago here: https://github.com/freeCodeCamp/freeCodeCamp/issues/50757

Ksound22 replied:
Shaun has confirmed it.

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c48df8674cf2b91020ecb.md:
````diff
@@ -0,0 +1,52 @@
+assert(document.querySelector('meta')?.getAttribute('charset')?.toLowerCase() === 'utf-8');
+```
+
+You should add a single `title` element.
````
Hey @zairahira 

Check this for an explanation from @ShaunSHamilton: https://github.com/freeCodeCamp/freeCodeCamp/issues/50757

jdwilkin4 replied:
@Ksound22 

Has this been resolved?
Is it cool if we close this conversation?

Ksound22 replied:
I will go ahead and close this and the rest

zairahira commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/64a2687ef267e5934a2f93e3.md:
```diff
@@ -0,0 +1,218 @@
+
+Using a descendant selector, select the two `div` elements inside the `div` with class `cat-mouth`. Give it a width of `30px`, a height of `50px`, and a border of `2px solid #000`.
+
+# --hints--
```
Some more variations that would be good to include:

```css
    .cat-mouth > cat-mouth-line-left, .cat-mouth > cat-mouth-line-right {
    
    /* code here */
    
    }
         
    
    .cat-mouth cat-mouth-line-left, .cat-mouth cat-mouth-line-right {
    
      /* code here */
    
    }
    
     
    .cat-mouth > div, .cat-mouth > div {
    
    /* code here */ 
    
    }
    
    
    .cat-mouth div, .cat-mouth div {
    
    /* code here */
    
    }
```

Ksound22 replied:
THis also works:
```css
.cat-mouth div {
    
    }
```

Especially for the whiskers `div` elements which have three `div` elements inside.

Is that okay by you?

zairahira commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f09293eb3230723a62f77.md:
```diff
@@ -0,0 +1,234 @@
+
+# --description--
+
+Access your `cat-mouth-line-right` element with a class selector, then move it into the right position with a `position` of `absolute`, a `top` of `88px` and a `right` of `91px`.
```
```suggestion
Access your `cat-mouth-line-right` element with a class selector, then move it into the right position with a `position` of `absolute`, a `top` of `88px` and a `left` of `91px`.
```

zairahira reviewed (commented):
@Ksound22  Reviewed all the steps, just left a few comments.

I also went through issue #50757 , seems like we can skip the checks for closing tags for now.

Ksound22 committed:
chore: implemented review suggestions from Dario

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c59e942f35541923104bf.md:
````diff
@@ -0,0 +1,66 @@
+You should have a `main` element.
+
+```js
+assert(document.querySelectorAll('main').length === 1);
````
@Ksound22 

Have the changes here been resolved?
If so, can we close this conversation

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c59e942f35541923104bf.md:
````diff
@@ -0,0 +1,66 @@
+```
+
+Your `div` element should have the class `cat-head`.
+
````
@Ksound22 

Have the changes here been resolved?
If so, can we close this conversation

jdwilkin4 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646c5ffef5598d449b52ec12.md:
````diff
@@ -0,0 +1,80 @@
+```js
+assert(document.querySelectorAll('.cat-head div')[0]?.getAttribute('class') === 'cat-ears');
+```
+
````
@Ksound22 

Have the changes here been resolved?
If so, can we close this conversation

jdwilkin4 reviewed (commented):
Hey @Ksound22 ! 

I think the project looks good.
I just wanted to double check on the open conversations and if they are resolved.

If all of those conversations are resolved and the changes have been applied then I am happy to approve 👍

Ksound22 committed:
chore: implemented a suggestion from Zaira

Co-authored-by: Zaira <33151350+zairahira@users.noreply.github.com>

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f102bf87b350b593baa72.md:
````diff
@@ -0,0 +1,256 @@
+assert(document.querySelectorAll('.cat-whiskers-right div')[0].classList.contains('cat-whisker-right-top'))
+```
+
+The second `div` element inside the `.cat-whiskers-right` element should have the class `cat-whisker-tight-middle`.
````
```suggestion
The second `div` element inside the `.cat-whiskers-right` element should have the class `cat-whisker-right-middle`.
```

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f135eab69d90d0c6d4e9b.md:
```diff
@@ -0,0 +1,254 @@
+
+# --description--
+
+Rotate the left top whisker at 10 degrees.
```
```suggestion
Rotate the left top whisker at `10` degrees.
```

naomi-lgbt commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646f1802a09a171332e14630.md:
```diff
@@ -0,0 +1,267 @@
+
+# --description--
+
+Rotate the left bottom whisker at -3 degrees.
```
```suggestion
Rotate the left bottom whisker at `-3` degrees.
```

naomi-lgbt reviewed (changes requested):
Hey Kolade, thanks for all of your hard work on this! It's looking great.

I've added a couple of formatting nitpicks - once those are resolved, happy to approve + merge.

Ksound22 commented on curriculum/challenges/english/14-responsive-web-design-22/learn-intermediate-css-by-building-a-cat-painting/646df0cf26413a7b35e4b8b3.md:
```diff
@@ -0,0 +1,210 @@
+
+Remove the sharp edges of the nose with border radius of `50%` each on the `top-left`, `bottom-right`, and `bottom-left` corners. Also, rotate it at 180 degrees.
+
+# --hints--
```
I think that makes sense to do. I will do it.

Ksound22 committed:
chore: implement review suggestions from Naomi

Co-authored-by: Naomi Carrigan <nhcarrigan@gmail.com>

Ksound22 committed:
feat: added a test to check for border-top-right-radius property in step 56

naomi-lgbt reviewed (approved):
Nice work on this Kolade! 💜
